### PR TITLE
feat(Live): add live snapshot resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add live snapshot resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add live snapshot resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/live/' TESTARGS='-run TestAccLiveSnapshot_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/live/ -v -run TestAccLiveSnapshot_basic -timeout 360m -parallel 4 
=== RUN   TestAccLiveSnapshot_basic 
=== PAUSE TestAccLiveSnapshot_basic
=== CONT  TestAccLiveSnapshot_basic
--- PASS: TestAccLiveSnapshot_basic (323.87s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/live      323.928s
```

### CODE REVIEW

File: huaweicloud/services/live/resource_huaweicloud_live_snapshot.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Live snapshot")
+	}
+
+	getLiveSnapshotRespBody, err := utils.FlattenResponse(getLiveSnapshotResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	snapshots := utils.PathSearch("snapshot_config_list", getLiveSnapshotRespBody,
```
Comment (hmbig2) [inline huaweicloud/services/live/resource_huaweicloud_live_snapshot.go]:
```suggestion
	snapshot := utils.PathSearch("snapshot_config_list|[0]", getLiveSnapshotRespBody, nil)
```

Comment (houpeng80) [reply huaweicloud/services/live/resource_huaweicloud_live_snapshot.go]: done

File: docs/resources/live_snapshot.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+---
+subcategory: "Live"
+---
+
+# huaweicloud_live_snapshot
+
+Manages a Live snapshot within HuaweiCloud.
```
Comment (deer-hang) [inline docs/resources/live_snapshot.md]:
```suggestion
Manages a Live snapshot resource within HuaweiCloud.
```

File: docs/resources/live_snapshot.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+variable "storage_path" {}
+variable "domain_name" {}
+
+resource "huaweicloud_live_snapshot" "test"{
+  domain_name    = var.domain_name
+  app_name       = "live"
+  frequency      = 10
+  storage_mode   = "0"
+  storage_bucket = var.storage_bucket
+  storage_path   = var.storage_path
```
Comment (deer-hang) [inline docs/resources/live_snapshot.md:22]: make the variable field together

File: huaweicloud/services/live/resource_huaweicloud_live_snapshot.go [hunk 4]
```diff
@@ -0,0 +1,335 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product Live
+// ---------------------------------------------------------------
+
+package live
```
Comment (deer-hang) [inline huaweicloud/services/live/resource_huaweicloud_live_snapshot.go:6]: remove the title

File: huaweicloud/services/live/resource_huaweicloud_live_snapshot.go [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+	}
+	_, err = deleteLiveSnapshotClient.Request("DELETE", deleteLiveSnapshotPath, &deleteLiveSnapshotOpt)
+	if err != nil {
+		return diag.Errorf("error deleting Live snapshot: %s", err)
+	}
+
+	return nil
+}
+
+func buildDeleteLiveSnapshotDeleteParams(d *schema.ResourceData) string {
```
Comment (deer-hang) [inline huaweicloud/services/live/resource_huaweicloud_live_snapshot.go]: `dimain_name` and `app_name` is forceNew and required. Ithink we can remove the `d.GetOK()` check. 
Just return `return fmt.Sprintf("?domain=%v&app_name=%v", d.Get("domain_name"), d.Get("app_name"))`

File: docs/resources/live_snapshot.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/live_snapshot.md]: done

File: huaweicloud/services/live/resource_huaweicloud_live_snapshot.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/live/resource_huaweicloud_live_snapshot.go]: fixed

**Pull request merged**